web/text_layer_builder.js

```diff
@@ -95,21 +116,50 @@ class TextLayerBuilder {
       textContent: this.textContent,
       textContentStream: this.textContentStream,
       container: textLayerFrag,
-      viewport: this.viewport,
+      viewport,
       textDivs: this.textDivs,
+      textDivProperties: this.textDivProperties,
       textContentItemsStr: this.textContentItemsStr,
+      isOffscreenCanvasSupported: this.isOffscreenCanvasSupported,
     });
-    this.textLayerRenderTask.promise.then(
-      () => {
-        this.textLayerDiv.append(textLayerFrag);
-        this.#finishRendering();
-        this.highlighter?.enable();
-        this.accessibilityManager?.enable();
-      },
-      function (reason) {
-        // Cancelled or failed to render text layer; skipping errors.
-      }
-    );
+
+    await this.textLayerRenderTask.promise;
+    this.div.append(textLayerFrag);
+    this.#finishRendering();
+    this.#scale = scale;
+    this.accessibilityManager?.enable();
+    this.show();
+  }
+
+  #setDimensions(viewport) {
+    const { div } = this;
+    const [pageLLx, pageLLy, pageURx, pageURy] = viewport.viewBox;
+    const pageWidth = pageURx - pageLLx;
+    const pageHeight = pageURy - pageLLy;
+    const { style } = div;
+
+    style.width = `calc(var(--scale-factor) * ${pageWidth}px)`;
+    style.height = `calc(var(--scale-factor) * ${pageHeight}px)`;
+  }
+
+  #update(viewport) {
+    const { rotation } = viewport;
+    if (rotation !== this.#rotation) {
+      this.#rotation = rotation;
+      this.div.setAttribute("data-main-rotation", rotation);
+    }
+  }
```

Comment:
*Back to something that I'm pretty sure that I've tried to mentioned before:*
By placing this code at the viewer level, rather than in the `src/text_layer.js` file, you've effectively made the "raw" textLayer-functionality that's exposed directly in the API *unusable* on its own (without a full viewer). This is clear regression that I'm pretty sure that some users will notice/complain about.

In the annotationLayer-implementation similar code is found in https://github.com/mozilla/pdf.js/blob/fa54a587906100eb8198de2e95d6b1e0bfa88846/src/display/annotation_layer.js#L2680-L2694, which means that it's still generally functional/usable.

Reply:
Also, the fact that the `setPageProperties`-method was necessary to add in `test/driver.js` feels like a good demonstration of how the basic textLayer-functionality is no longer directly usable now.